modules/ssh/ssh.go

```diff
@@ -75,11 +75,21 @@ func sessionHandler(session ssh.Session) {
 	ctx, cancel := context.WithCancel(session.Context())
 	defer cancel()
 
+	gitProtocol := ""
+	for _, env := range session.Environ() {
+		if strings.HasPrefix(env, "GIT_PROTOCOL=") {
+			// The value would be version=2, so using normal split doesn't work here.
+			gitProtocol = strings.SplitN(env, "=", 2)[1]
```

Comment:
```go
		_, gitProtocol, _ = strings.Cut(env, "=")
```

Reply:
strings.Cut was introduced in Go1.18, Gitea's minimum go version is still 1.17

Reply:
Gitea uses go 1.18 as of https://github.com/go-gitea/gitea/pull/19918

Reply:
Huh? I was under the impression it was always minimum version is stable-1

Reply:
We decided to switch a bit earlier to 1.18.

As now 1.19 is out, we are defintely back on schedule.

Anyway, this discussion is probably a little out of place here...

Reply:
https://github.com/go-gitea/gitea/pull/20638